main/src/main/scala/sbt/RemoteCache.scala

```diff
@@ -78,43 +65,6 @@ object RemoteCache {
         val tempDiskCache = (s.baseDir / "target" / "bootcache").toPath()
         Def._cacheStore = DiskActionCacheStore(tempDiskCache)
 
-  private[sbt] def getCachedAnalysis(ref: String): CompileAnalysis =
-    getCachedAnalysis(CacheImplicits.strToHashedVirtualFileRef(ref))
-  private[sbt] def getCachedAnalysis(ref: HashedVirtualFileRef): CompileAnalysis =
-    analysisStore.getOrElseUpdate(
-      ref, {
-        val vfs = cacheStore.getBlobs(ref :: Nil)
-        if vfs.nonEmpty then
-          val outputDirectory = Def.cacheConfiguration.outputDirectory
-          cacheStore.syncBlobs(vfs, outputDirectory).headOption match
-            case Some(file) => FileAnalysisStore.binary(file.toFile()).get.get.getAnalysis
-            case None       => Analysis.empty
-        else Analysis.empty
-      }
-    )
-
-  private[sbt] val tempConverter: MappedFileConverter = MappedFileConverter.empty
-  private[sbt] def postAnalysis(analysis: CompileAnalysis): Option[HashedVirtualFileRef] =
-    IO.withTemporaryFile("analysis", ".tmp", true): file =>
-      val output = CompileOutput.empty
-      val option = MiniOptions.of(Array(), Array(), Array())
-      val setup = MiniSetup.of(
-        output,
-        option,
-        "",
-        xsbti.compile.CompileOrder.Mixed,
-        false,
-        Array()
-      )
-      FileAnalysisStore.binary(file).set(AnalysisContents.create(analysis, setup))
-      val vf = tempConverter.toVirtualFile(file.toPath)
-      val refs = cacheStore.putBlobs(vf :: Nil)
-      refs.headOption match
-        case Some(ref) =>
-          analysisStore(ref) = analysis
-          Some(ref)
-        case None => None
-
```

Comment:
We don't need the cache to post and get the analyses of the classpath. First because they are already stored by the `compile` task and second because we can directly read the analyses in the target directories of each module.

Reply:
If I remember correctly, I think my motivation here was to unify the machine-wide disk caching with analysis cache. In addition to clean compile, and resuming from the fully compiled state, there's no a third scenario where a disk cache (or a remote cache) is hydrated but local `target/` is empty. If there's a race, probably we need a way for the participants to coordinate better?

Reply:
> If I remember correctly, I think my motivation here was to unify the machine-wide disk caching with analysis cache.

I am not sure about that because `postAnalysis`  and `getCachedAnalysis` were only used to create instances of `Classpath` and to extract `CompileAnalysis` from `Classpath`.

> In addition to clean compile, and resuming from the fully compiled state, there's no a third scenario where a disk cache (or a remote cache) is hydrated but local target/ is empty.

I am not sure to understand what you mean here. When we get a `Classpath`, we know that `compile` was executed and that the analyses were produced, or retrieved from the cache.

> If there's a race, probably we need a way for the participants to coordinate better?

Yes but we have the task engine to coordinate. Accessing the cache is a side effect and as such it should be managed by the task engine. In that sense, `Def.cachedTask` is super safe because it ensures that a single task is responsible from posting or getting the same file from the cache. As soon as we bypass the task engine and access the cache directly we lose those guarantees.

Also the `IO.withTemporaryFile("analysis", ".tmp", true)` is suspicious because it creates randomness. It makes it impossible to hash any instance of `Classpath` deterministically.